core/src/main/scala/org/bykn/bosatsu/SourceConverter.scala

```diff
@@ -578,6 +576,8 @@ final class SourceConverter(
             else success(neBound.toList ::: missingFromDecl.map((_, None)))
         }
 
+    // TODO we have to make sure we don't have more than 8 arguments to a struct
```

Comment:
make sure to use `FnType.isValid` or whatever so we can set the function maximum size in one place.

Reply:
remove this TODO